### PR TITLE
slight tweaks made while replacing squares with chessground in mobile

### PR DESCRIPTION
**Pull request by schlawg**
Hi Vincent, I don't know if you're seeking pull requests at this early stage but I was replacing squares & bishop with your chessground in a mobile (tv stream) prototype and made some changes that simplify client code.  Basically, a make from uci string static in Move and also made Move.promotion a PieceRole rather than a Piece.

### CODE REVIEW

**Comment by veloce** [comment]
Indeed it is still early stage and it has not been tested in a real app yet. Your change seems legit as it simplifies client code. Can you post as well a small snippet where you show the resulting change in client code? Thanks.

**Comment by schlawg** [comment]
Sure.  Here's chessground used as a stateless renderer for tv stream events:

```
                  return Container(
                    margin: const EdgeInsets.all(8.0),
                    padding: const EdgeInsets.all(8.0),
                    decoration: BoxDecoration(border: Border.all()),
                    child: Column(
                      children: [
                        _playerInfo(state.players.last),
                        Board(
                            key: GlobalKey(),
                            size: 300,
                            orientation: Color.white,
                            theme: BoardTheme.blue,
                            fen: state.fen,
                            lastMove: Move.make(state.lastMove)),
                        _playerInfo(state.players.first),
                        const SizedBox(height: 10),
                      ],
                    ),
```
Before the change, I interpreted the lastMove json string, figured out the current color, and created a piece of that color for promotions in order to construct a Move to properly instantiate the lastMove field in the board constructor.  Now it's done in Move.make.

If this stateless static display is an approved usage, it would also be nice if there was an animateLastMove boolean in Board constructor.  

P.S. The uci string stuff in Move was before I noticed fen.dart.  Maybe those switches can be replaced with or moved into fen.

File: lib/src/models.dart
Comment (veloce) [inline lib/src/models.dart:166]: You should use a [named constructor](https://dart.dev/guides/language/language-tour#named-constructors) instead of a static method, and name it `Move.fromUci(String uci)` so it's consistent with `Coord.fromSquareId`. `uci` param should not be optional

**Review by veloce — CHANGES_REQUESTED** [review]
Thanks for this! I need just this one change before merging.

**Comment by veloce** [comment]
I renamed `Move.make` to `Move.fromUci` for more clarity.

**Pull request merged**
